py-polars/polars/datatypes/classes.py

```diff
@@ -813,7 +813,7 @@ def __init__(
             self.size = shape
             self.shape = (shape,) + inner_shape
 
-        elif isinstance(shape, tuple):
+        elif isinstance(shape, tuple) and isinstance(shape[0], int):  # type: ignore[redundant-expr]
```

Comment:
Adding type ignore, otherwise mypy raises the following:

> error: Left operand of "and" is always true  [redundant-expr]

I assume due to the type annotation in the signature